sphinx/writers/latex.py

```diff
@@ -572,6 +572,19 @@ def __init__(self, document, builder):
         else:
             self.elements['date'] = format_date(builder.config.today_fmt or _('%b %d, %Y'),  # type: ignore  # NOQA
                                                 language=builder.config.language)
+
+        if builder.config.numfig:
+            self.numfig_secnum_depth = builder.config.numfig_secnum_depth
+            if self.numfig_secnum_depth > 0:  # default is 1
+                self.numfig_secnum_depth += self.top_sectionlevel - 1
+                if self.numfig_secnum_depth >= len(self.sectionnames):
+                    self.numfig_secnum_depth = len(self.sectionnames) - 1
+                if self.numfig_secnum_depth >= 0:
+                    self.elements['sphinxpkgoptions'] += \
+                    (',numfigreset=%s' % self.sectionnames[self.numfig_secnum_depth])
+            else:
+                self.elements['sphinxpkgoptions'] += ',nonumfigreset'
+
```

Comment:
in this way if numfig=False, neither ``nonumfigreset`` nor ``numfigreset=<foo>`` is used, and the LaTeX code executes exactly as prior to this PR for backwards compatibility.